internal/kubernetes/unauthenticated.go

```diff
@@ -35,7 +37,19 @@ func MakeUnauthenticatedRequest(ctx context.Context, endpoint string, caCertific
 		return validation.WithRemediation(err, "Ensure the Kubernetes API server endpoint provided is correct.")
 	}
 
-	resp, err := client.Do(req)
+	consecutiveErrors := 0
+	var resp *http.Response
+	err = wait.PollUntilContextTimeout(ctx, validation.ValidationInterval, validation.ValidationTimeout, true, func(ctx context.Context) (bool, error) {
```

Comment:
you might want to extend this loop to line 59. what if you dial in correctly but fail reading the rest of the response? the network connection can be interrupted at any point of the request